com.eclipsesource.modelserver.emf/src/main/java/com/eclipsesource/modelserver/emf/common/ModelRepository.java

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+ * Copyright (c) 2019 EclipseSource and others.
+ *
+ *   This program and the accompanying materials are made available under the
+ *   terms of the Eclipse Public License v. 2.0 which is available at
+ *   http://www.eclipse.org/legal/epl-2.0.
+ *
+ *   This Source Code may also be made available under the following Secondary
+ *   Licenses when the conditions for such availability set forth in the Eclipse
+ *   Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ *   with the GNU Classpath Exception which is available at
+ *   https://www.gnu.org/software/classpath/license.html.
+ *
+ *   SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ *******************************************************************************/
+package com.eclipsesource.modelserver.emf.common;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
+
+import com.eclipsesource.modelserver.emf.ResourceManager;
+import com.eclipsesource.modelserver.emf.configuration.ServerConfiguration;
+import com.google.inject.Inject;
+
+/**
+ * Injectable singleton class represents a repository of all loaded models and provides a CRUD API.
+ *
+ */
+public class ModelRepository {
+	
+	private Map<URI, EObject> models;
+	
+	@Inject
+	private ServerConfiguration serverConfiguration;
+	@Inject
+	private ResourceManager resourceManager;
+
+	private ResourceSet resourceSet = new ResourceSetImpl();
+
+	@Inject
+	public ModelRepository() {
+		this.models = new HashMap<URI, EObject>();
```

Comment:
No need for type specifier on RHS

Reply:
I would prefer to keep type specifiers, to reduce the number of warnings.

Reply:
Why are you getting warnings?

Reply:
- HashSet is a raw type. References to generic type HashSet<E> should be parameterized
- Type safety: The expression of type HashSet needs unchecked conversion to conform to Set<String>

Am I missing some setting?

Reply:
Sounds like you were using  `new HashSet` instead of `new HashSet<>`

Reply:
I don't have a strong opinion on that so I have merged it anyway.

Reply:
Ah thanks! Will remove the specifiers next time then!